src/arena/CreateSprintTemplate.tsx

```diff
@@ -91,7 +91,7 @@ function CreateSprintTemplate(props: CreateSprintTemplateProps) {
   const { handleShowError, handleShowSuccess } = useContext(ToastMsgContext);
 
   const theme = useTheme();
-  const navigate = useNavigate();
+  const navigate = props.navigate;
```

Comment:
Just a random question, why the change to props.navigate instead of usenavigate hook?

Reply:
It's mainly to make unit testing easier. If we pass the navigation function as a prop, then we can easily mock it and prevent it from throwing an error. 

useNavigate can only be used in the context of a Router, so if we render the component for testing _without_ a router context, the test will fail.  Calling useNavigate() will throw an error and prevent the component from rendering. 

Another alternative would be to wrap the rendered component in a generic router component in the test file, ie:

```typescript
// CreateSprintTemplate.spec.tsx

62  render(
63    <BrowserRouter>
64      <CreateSprintTemplate
65       user={testUser}
66        navigate={mockNavigate}
67        getTemplates={mockGetTemplates}
68        setTemplate={mockSetTemplate}
69        getTemplateOptionsFromSanity={mockGetTemplateOptions}
70      />
71    </BrowserRouter>
72  );
```
Definitely open to debate on which approach is better. I personally prefer the simplicity of just rendering the specific component in the test, but the trade-off is a bit more complexity in how the component is set up and what needs to be mocked in the unit tests.

Though, thanks for mentioning it - I took a second look and how I did it needed to be tweaked. It wouldn't cause issues with how the repository is currently set up, but if we get fancier with navigation in the future it could cause issues.

Eventually I'll probably switch to the strategy discussed here: https://blog.logrocket.com/testing-react-router-usenavigate-hook-react-testing-library/ or I might put it up as an issue for someone to work on.

Reply:
Awesome, I just wanted to know your thinking, because I didnt know any better :)